src/utils/update/psc.to.beneficial.owner.type.mapper.ts

```diff
@@ -71,91 +73,190 @@ export const mapPscToBeneficialOwnerGov = (psc: CompanyPersonWithSignificantCont
+  beneficialOwner.owner_of_land_other_entity_nature_of_control_jurisdictions = [];
 
   psc.naturesOfControl?.forEach(natureType => {
-    const controlKind = natureTypeMap.get(natureType);
-    if (!controlKind) {
+    const natureKind = natureTypeMap.get(natureType);
+    if (!natureKind) {
       return;
     }
-    const natureOfControlType = natureOfControlTypeMap.get(natureType);
 
+    const natureOfControlType = natureOfControlTypeMap.get(natureType);
     if (natureOfControlType) {
-      switch (controlKind) {
-          case 'BoNatureOfControl':
-            beneficialOwner.beneficial_owner_nature_of_control_types?.push(natureOfControlType);
-            break;
-          case "NonLegalNatureOfControl":
-            beneficialOwner.non_legal_firm_members_nature_of_control_types?.push(natureOfControlType);
-            break;
-          case "TrustNatureOfControl":
-            if (!isBeneficialGov){
-              beneficialOwner['trustees_nature_of_control_types'].push(natureOfControlType);
-            }
-            break;
-          default:
-            logger.error("Unexpected nature of control for BO: " + controlKind);
-            break;
-      }
+      addNatureOfControlTypeToBeneficialOwner(natureKind, beneficialOwner, natureOfControlType, isBeneficialGov);
+      return;
+    }
+
+    const natureOfControlJurisdiction = natureOfControlJurisdictionMap.get(natureType);
+    if (natureOfControlJurisdiction) {
+      addNatureOfControlJurisdictionToBeneficialOwner(natureKind, beneficialOwner, natureOfControlJurisdiction);
     }
   });
 };
 
+const addNatureOfControlTypeToBeneficialOwner = (natureKind: BoTypes, beneficialOwner: BeneficialOwnerType, natureOfControlType: NatureOfControlType, isBeneficialGov: boolean) => {
+  switch (natureKind) {
+      case BoTypes.BO_NATURE_OF_CONTROL:
+        beneficialOwner.beneficial_owner_nature_of_control_types?.push(natureOfControlType);
+        break;
+      case BoTypes.NON_LEGAL_NATURE_OF_CONTROL:
+        beneficialOwner.non_legal_firm_members_nature_of_control_types?.push(natureOfControlType);
+        break;
+      case BoTypes.TRUST_NATURE_OF_CONTROL:
+        if (!isBeneficialGov) {
+          beneficialOwner['trustees_nature_of_control_types'].push(natureOfControlType);
+        }
+        break;
+      case BoTypes.CONTROL_OVER_TRUST_NATURE_OF_CONTROL:
+        beneficialOwner.trust_control_nature_of_control_types?.push(natureOfControlType);
+        break;
+      case BoTypes.CONTROL_OVER_FIRM_NATURE_OF_CONTROL:
+        beneficialOwner.non_legal_firm_control_nature_of_control_types?.push(natureOfControlType);
+        break;
+      default:
+        logger.error("Unexpected nature of control for BO: " + natureKind);
+        break;
+  }
+};
+
+const addNatureOfControlJurisdictionToBeneficialOwner = (natureKind: BoTypes, beneficialOwner: BeneficialOwnerType, natureOfControlJurisdiction: NatureOfControlJurisdiction) => {
+  switch (natureKind) {
```

Comment:
All the cases do the same operation, so you can simplify the code by doing something similar to this:

`
enum NatureOfControlJurisdictionEnum {
  "RegisteredOwnerAsNomineePersonNatureOfControl" = "owner_of_land_person_nature_of_control_jurisdictions",
  "RegisteredOwnerAsNomineeOtherEntityNatureOfControl" = "owner_of_land_other_entity_nature_of_control_jurisdictions",
}

const addNatureOfControlJurisdictionToBeneficialOwner = (natureKind: BoTypes, beneficialOwner: BeneficialOwnerType, natureOfControlJurisdiction: NatureOfControlJurisdiction) => {

  const res = NatureOfControlJurisdictionEnum[natureKind];

  if (!res) {
    logger.error("Unexpected nature of control for BO: " + natureKind);
    return;
  }

  beneficialOwner[res]?.push(natureOfControlJurisdiction);
};
`

same for the one above

Reply:
updated